Vowpal Wabbit - Contextual Bandits.ipynb

```diff
@@ -0,0 +1,705 @@
+      "Input label = cb\n",
+      "Output pred = action_scores\n",
+      "average  since         example        example        current        current  current\n",
+      "loss     last          counter         weight          label        predict features\n",
+      "0.000000 0.000000            1            1.0      9:-1:0.82            0:0     1791\n",
+      "\n",
+      "finished run\n",
+      "number of examples = 1\n",
+      "weighted example sum = 1.000000\n",
+      "weighted label sum = 0.000000\n",
+      "average loss = 0.000000\n",
+      "total feature number = 1726\n",
+      "0.000000 0.000000            2            2.0      9:-1:0.82        6:-2.67     1791\n",
+      "\n",
+      "finished run\n",
+      "number of examples = 2\n",
+      "weighted example sum = 2.000000\n",
+      "weighted label sum = 0.000000\n",
+      "average loss = 0.000000\n",
+      "total feature number = 3452\n",
+      "23/01/16 13:53:11 WARN VowpalWabbitGeneric: VowpalWabbit args: --cb_adf --cb_type mtr --clip_p 0.1 -q GT -q MS -q GR -q OT -q MT -q OS --dsjson --preserve_performance_counters --quiet --no_stdin)\n"
+     ]
+    }
+   ],
+   "source": [
+    "from synapse.ml.vw import VowpalWabbitGeneric\n",
+    "\n",
+    "model = (\n",
+    "    VowpalWabbitGeneric(\n",
+    "        passThroughArgs=\"--cb_adf --cb_type mtr --clip_p 0.1 -q GT -q MS -q GR -q OT -q MT -q OS --dsjson --preserve_performance_counters\"\n",
+    "    )\n",
+    "    .setInputCol(\"input\")\n",
+    "    .setSplitCol(\"splitId\")\n",
+    "    .setPredictionIdCol(\"EventId\")\n",
+    "    .fit(df_ready)\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Model Prediction"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 39,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "root\n",
+      " |-- EventId: string (nullable = true)\n",
+      " |-- input: string (nullable = true)\n",
+      " |-- json: struct (nullable = true)\n",
+      " |    |-- EventId: string (nullable = true)\n",
+      " |    |-- _label_probability: float (nullable = true)\n",
+      " |    |-- _labelIndex: integer (nullable = true)\n",
+      " |    |-- _label_cost: float (nullable = true)\n",
+      " |-- rewards: struct (nullable = false)\n",
+      " |    |-- reward: float (nullable = true)\n",
+      " |-- probLog: float (nullable = true)\n",
+      " |-- chosenActionIndex: integer (nullable = true)\n",
+      " |-- splitId: integer (nullable = false)\n",
+      " |-- predictions: array (nullable = true)\n",
+      " |    |-- element: struct (containsNull = true)\n",
+      " |    |    |-- action: integer (nullable = false)\n",
+      " |    |    |-- score: float (nullable = false)\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "df_headers_predictions.printSchema()"
```

Comment:
Probabbly dont need the calls to print schema everywhere as the notebook outputs will be cleaned

Reply:
The reward column is nested and really hard to see from just display. added comment.